### PR TITLE
Sentry integration

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/335)

### CODE REVIEW

File: integrations/Sentry/browser.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  // eslint-disable-next-line class-methods-use-this
+  isLoaded() {
+    logger.debug("===in Sentry isLoaded===");
+    return !!(window.Sentry && window.Sentry.setUser !== Array.prototype.push);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  isReady() {
+    logger.debug("===in Sentry isReady===");
+    return !!(window.Sentry && window.Sentry.setUser !== Array.prototype.push);
```
Comment (arajguha) [inline integrations/Sentry/browser.js:56]: `message` is not defined

Comment (arajguha) [inline integrations/Sentry/browser.js:56]: can we check like this? `return !!(window.Sentry && window.Sentry.setUser)`. Why exactly are we comparing it with `Array.prototype.push`

File: integrations/Sentry/utils.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  js.type = "text/javascript";
+  js.id = id;
+  const e = document.getElementsByTagName("script")[0];
+  logger.debug("==parent script==", e);
+  logger.debug("==adding script==", js);
+  e.parentNode.insertBefore(js, e);
+};
+
+const identifierPayloadBuilder = (userId, email, name, ip_address) => {
+    payload = {};
```
Comment (arajguha) [inline integrations/Sentry/utils.js:19]: `payload` is not defined

File: integrations/Sentry/utils.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import logger from "../../utils/logUtil";
+
+const convertObjectToArray = (objectInput) => {
+  const list = [];
+  objectInput.forEach(function (objectItem) {
```
Comment (arajguha) [inline integrations/Sentry/utils.js:5]: let's use es6 arrow functions

File: integrations/Sentry/utils.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const x = objectItem.eventCustomProperties;
+    list.push(x);
+  });
+  return list;
```
Comment (arajguha) [inline integrations/Sentry/utils.js:9]: can we do this in oneliner? 
eg: `return objectInput.map(objectItem => objectItem.eventCustomProperties)`

File: integrations/Sentry/utils.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  logger.debug("==parent script==", e);
+  logger.debug("==adding script==", js);
+  e.parentNode.insertBefore(js, e);
+  SentryRewriteFramesLoader(
+    "Sentry",
+    `https://browser.sentry-cdn.com/6.12.0/rewriteframes.min.js`
+  );
+};
+
+const identifierPayloadBuilder = (userId, email, name, ipAddress) => {
```
Comment (arajguha) [inline integrations/Sentry/utils.js:48]: can we refactor this function?

File: integrations/Sentry/utils.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    "sha384-WOm9k3kzVt1COFAB/zCXOFx4lDMtJh/2vmEizIwgog7OW0P/dPwl3s8f6MdwrD7q";
+  js.crossorigin = "anonymous";
+  js.async = true;
+  js.type = "text/javascript";
+  js.id = id;
+  const e = document.getElementsByTagName("script")[0];
+  logger.debug("==parent script==", e);
+  logger.debug("==adding script==", js);
+  e.parentNode.insertBefore(js, e);
+};
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply integrations/Sentry/utils.js:26]: done

File: integrations/Sentry/browser.js [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+          regex = new RegExp(path);
+        } catch (e) {
+          // ignored
+        }
+        return regex;
+      });
+    }
+
+    if (includePaths.length > 0) {
+      this.integrations.push(
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply integrations/Sentry/browser.js:76]: made it `window.Sentry.integrations.push(`

File: integrations/Sentry/browser.js [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+        new window.Sentry.Integrations.RewriteFrames({
+          // eslint-disable-next-line object-shorthand
+          iteratee: function (frame) {
+            // eslint-disable-next-line consistent-return
+            includePaths.forEach((i) => {
+              try {
+                if (frame.filename.match(includePaths[i])) {
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply integrations/Sentry/browser.js:83]: corrected

File: integrations/Sentry/utils.js (hunk 4 above)
Comment (shrouti1507) [reply integrations/Sentry/utils.js:9]: done

**Review by saikumarrs — APPROVED** [review]

**Review by arajguha — APPROVED** [review]

**Pull request merged**